Sources/protoc-gen-swift/MessageGenerator.swift

```diff
@@ -250,61 +250,25 @@ class MessageGenerator {
     }
 
     private func generateProtoNameProviding(printer p: inout CodePrinter) {
-        if descriptor.reservedNames.isEmpty && descriptor.reservedRanges.isEmpty {
-            if fields.isEmpty {
-                p.print("\(visibility)static let _protobuf_nameMap = \(namer.swiftProtobufModulePrefix)_NameMap()")
-            } else {
-                p.print("\(visibility)static let _protobuf_nameMap: \(namer.swiftProtobufModulePrefix)_NameMap = [")
-                p.withIndentation { p in
-                    for f in fields {
-                        for n in f.fieldMapNames {
-                            p.print("\(f.number): \(n),")
-                        }
-                    }
-                }
-                p.print("]")
-            }
+        var writer = ProtoNameInstructionWriter()
+        for f in fields {
+            f.writeProtoNameInstruction(to: &writer)
+        }
+        for name in descriptor.reservedNames {
+            writer.writeReservedName(name)
+        }
+        for range in descriptor.reservedRanges {
+            writer.writeReservedNumbers(range)
+        }
+        if fields.isEmpty && descriptor.reservedNames.isEmpty && descriptor.reservedRanges.isEmpty {
```

Comment:
Rather than checking the fields and reserved bits, does it make sense to expose something on the writer to detect if it is empty except for the version marker? `writer.UseEmptyInitializer` or something like that?